src/services/resource-service/resource.service.ts

```diff
@@ -0,0 +1,66 @@
+import {Injectable} from '@angular/core';
+import {LocalStorageService} from '../storage-service/local-storage-service.service';
+import {Project, PROJECT_KEY} from '../../interfaces/project';
+import {Observable} from 'rxjs';
+import {Resource} from '../../interfaces/resource';
+import {ProjectService} from '../project-service/project.service';
+import {tap} from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ResourceService {
+
+  constructor(private localStorageService: LocalStorageService, private projectService: ProjectService) {
+  }
+
+  private getResource(project: Project, name: string): Resource {
```

Comment:
a. This could be a getter
b. good job using find function
c. when you use a function in the html, angular will run it on every cycle. It is better to keep a variable with the relevant resource  and update it whenever there is a change, then to run over all the projects in each cycle